balancer/rls/cache_test.go

```diff
@@ -119,7 +120,7 @@ func (s) TestLRU_BasicOperations(t *testing.T) {
 
 func (s) TestDataCache_BasicOperations(t *testing.T) {
 	initCacheEntries()
-	dc := newDataCache(5, nil)
+	dc := newDataCache(5, nil, &stats.NoopMetricsRecorder{}, "")
```

Comment:
Is there a plan to add a metrics recorder type that can be used in unit tests, or does exist already? One that allows the test to validate values recorded on the recorder?

We should ideally use one such recorder here. If the plan is to add such a thing later on, or use an available thing later on, we should add a TODO here and file an issue for the same.

Reply:
Yeah, we do, here: https://github.com/grpc/grpc-go/blob/master/internal/testutils/stats/test_metrics_recorder.go#L38. I was actually scaling these unit tests up to handle both what you are testing and the metrics emitted. However, for WRR, Doug didn't like this being an extra assertion to his unit tests, and wanted a separate test for metrics emissions. Do you share the same concern with having this test test both a. the functionality you are testing and b. the metrics that are emitted?

Reply:
I'd prefer separate tests too. Thanks.

Reply:
Will do.